main/inc/lib/document.lib.php

```diff
@@ -6694,4 +6711,113 @@ public static function resizeImageSlideShow(
 
         return $result;
     }
+
+    /**
+     * Adds a cloud link to the database
+     *
+     * @author - Aquilino Blanco Cores <aqblanco@gmail.com>
+     * @param array $_course
+     * @param string $path
+     * @param string $url
+     * @return int id of document or 0 if already exists or there was a problem creating it
+     */
+    public static function addCloudLink($_course, $path, $url, $name)
+    {
+        require_once api_get_path(LIBRARY_PATH) . 'fileUpload.lib.php';
+        $file_path = $path;
+        if (!self::cloudLinkExists($_course, $path, $url)) {
+            $doc_id = add_document($_course, $file_path, "link", 0, $name, $url);
+            if ($doc_id) {
+                // Update document item_property
+                api_item_property_update($_course, TOOL_DOCUMENT, $doc_id, 'DocumentAdded', api_get_user_id(), api_get_group_id(), api_get_user_id(), null, null, api_get_session_id());
+            }
+    
+            // If the file is in a folder, we need to update all parent folders
+            item_property_update_on_folder($_course, $file_path, api_get_user_id());
+    
+            return $doc_id;
+        } else {
+            return 0;
+        }
+    }
+ 
+    /**
+     * Deletes a cloud link from the database
+     *
+     * @author - Aquilino Blanco Cores <aqblanco@gmail.com>
+     * @param array $_course
+     * @param string $id
+     * @param string $url
+     * @return boolean true if success / false if an error occurred
+     */
+    public static function deleteCloudLink($_course, $id)
+    {
+ 
+        if (empty($id)) {
+            return false;
+        }
+ 
+        $document_id = $id;
```

Comment:
variables used in queries *always* have to be filtered down to their minimum possible expression just before using them in the query. In this case, you *have* to make sure $id is integer (otherwise you are introducing an SQL injection):
```
$document_id = intval($id);
```

Reply:
Better to use:

``
$documentId = (int) $id;
``

- CamelCase variable name
- casting is faster than using intval

Reply:
Well yes for intval vs casting. I think $document_id is just copied from some other function though, but you're right, our convention has now been to use camelCase for all variable names for a few years. This being said, I would accept the PR event without this camelCase (style issue), but not without the conversion to int (security issue)

Reply:
Already fixed both issues